sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,358 @@
+                serviceCompletionHandler.removeCallbacks(serviceCompletionTimeoutCallback);
+
+                streamingStateMachine.transitionToState(StreamingStateMachine.ERROR);
+                Log.e(TAG, "OnServiceError: " + reason);
+                streamingStateMachine.transitionToState(StreamingStateMachine.NONE);
+
+                if (serviceCompletionListener != null) {
+                    CompletionListener completionListener = serviceCompletionListener;
+                    serviceCompletionListener = null;
+                    completionListener.onComplete(false);
+                }
+            }
+        }
+    };
+
+    /**
+     * Creates a new object of AudioStreamManager
+     * @param internalInterface The internal interface to the connected device.
+     */
+    public AudioStreamManager(@NonNull ISdl internalInterface, @NonNull Context context) {
+        super(internalInterface);
+        this.queue = new LinkedList<>();
+        this.context = new WeakReference<>(context);
+        this.serviceCompletionHandler = new Handler(Looper.getMainLooper());
+
+        internalInterface.addServiceListener(SessionType.PCM, serviceListener);
+
+        streamingStateMachine = new StreamingStateMachine();
+        transitionToState(BaseSubManager.READY);
+    }
+
+    @Override
+    public void dispose() {
+        stopAudioStream(new CompletionListener() {
+            @Override
+            public void onComplete(boolean success) {
+                internalInterface.removeServiceListener(SessionType.PCM, serviceListener);
+            }
+        });
+
+        super.dispose();
+    }
+
+    /**
+     * Starts the audio service and audio stream to the connected device.
+     * The method is non-blocking.
+     * @param encrypted Specify whether or not the audio stream should be encrypted.
+     */
+    public void startAudioStream(boolean encrypted, final CompletionListener completionListener) {
+        // audio stream cannot be started without a connected internal interface
+        if (!internalInterface.isConnected()) {
+            Log.w(TAG, "startAudioStream called without being connected.");
+            if (completionListener != null) {
+                completionListener.onComplete(false);
+            }
+            return;
+        }
+
+        // streaming state must be NONE (starting the service is ready. starting stream is started)
+        if (streamingStateMachine.getState() != StreamingStateMachine.NONE) {
+            Log.w(TAG, "startAudioStream called but streamingStateMachine is not in state NONE (current: " + streamingStateMachine.getState() + ")");
+            if (completionListener != null) {
+                completionListener.onComplete(false);
+            }
+            return;
+        }
+
+        AudioPassThruCapabilities capabilities = (AudioPassThruCapabilities) internalInterface.getCapability(SystemCapabilityType.PCM_STREAMING);
+
+        switch (capabilities.getSamplingRate()) {

```

Comment:
Would probably be good to add a `null` check to `capabilities`

Reply:
Done